cli.py

```diff
@@ -0,0 +1,161 @@
+            safe_mkdir(network_dir)
+            write_json(network_dir, f'{contract["contractType"]}.json', json.dumps(contract["abi"], indent = 2))
+
+
+        new_addressbook_dump = json.dumps(new_addressbook, indent = 2)
+        write_json(version_dir, "addressbook.minified.json", new_addressbook_dump)
+
+
+    def patch_address(self, version, contract_name, new_address):
+        time_now_gmt = time_now()
+        root_dir = os.getcwd()
+        version_dir = os.path.join(root_dir, version)
+
+        # Update legacy addressbook
+        legacy_addressbook_path = os.path.join(version_dir, "addressbook.json")
+        legacy_addressbook = read_json(legacy_addressbook_path)
+        if contract_name in legacy_addressbook:
+            legacy_addressbook[contract_name]["address"] = new_address
+            legacy_addressbook[contract_name]["lastUpdated"] = time_now_gmt
+        legacy_addressbook_dump = json.dumps(legacy_addressbook, indent = 2)
+        write_json(version_dir, "addressbook.json", legacy_addressbook_dump)
+
+        # Update new addressbook
+        new_addressbook_path = os.path.join(version_dir, "addressbook.minified.json")
+        new_addressbook = read_json(new_addressbook_path)
+        for network_id in new_addressbook.keys():
+            if contract_name in new_addressbook[network_id]:
+                new_addressbook[network_id][contract_name]["address"] = new_address
+                new_addressbook[network_id][contract_name]["lastUpdated"] = time_now_gmt
+        new_addressbook_dump = json.dumps(new_addressbook, indent = 2)
+        write_json(version_dir, "addressbook.minified.json", new_addressbook_dump)
+
+    def patch_abi(self, version, contract_type, contract_abi_repo_branch):
+        new_abi_address = f'https://raw.githubusercontent.com/AffineLabs/contracts/{contract_abi_repo_branch}/abi/{contract_type}.json'
+        new_abi = requests.get(new_abi_address)
+
+        time_now_gmt = time_now()
+        root_dir = os.getcwd()
+        version_dir = os.path.join(root_dir, version)
+
+        # Update legacy addressbook
+        legacy_addressbook_path = os.path.join(version_dir, "addressbook.json")
+        legacy_addressbook = read_json(legacy_addressbook_path)
+        for contract_name in legacy_addressbook.keys():
+            if legacy_addressbook[contract_name]["contractType"] == contract_type:
+                legacy_addressbook[contract_name]["abi"] = new_abi.json()
+                legacy_addressbook[contract_name]["lastUpdated"] = time_now_gmt
+        legacy_addressbook_dump = json.dumps(legacy_addressbook, indent = 2)
+        write_json(version_dir, "addressbook.json", legacy_addressbook_dump)
+
+        # Update new addressbook
+        new_addressbook_path = os.path.join(version_dir, "addressbook.minified.json")
+        new_addressbook = read_json(new_addressbook_path)
+        for network_id in new_addressbook.keys():
+            for contract_name in legacy_addressbook[network_id].keys():
```

Comment:
The legacy address book uses contract names as keys, not network ids. Correct?

Reply:
Fixed